File: src/vibeusage/config/paths.py
"""Platform-specific paths for vibeusage configuration and cache."""

from __future__ import annotations

import os
from pathlib import Path

from platformdirs import user_cache_dir
from platformdirs import user_config_dir
from platformdirs import user_state_path

PACKAGE_NAME = "vibeusage"


def _get_env_path(env_var: str, fallback: Path) -> Path:
    """Get path from environment variable or fallback."""
    if env_value := os.environ.get(env_var):
        return Path(env_value)
    return fallback


def config_dir() -> Path:
    """Get user config directory.

    Respects VIBEUSAGE_CONFIG_DIR environment variable.
    """
    base_dir = Path(user_config_dir(PACKAGE_NAME))
    return _get_env_path("VIBEUSAGE_CONFIG_DIR", base_dir)


def cache_dir() -> Path:
    """Get user cache directory.

    Respects VIBEUSAGE_CACHE_DIR environment variable.
    """
    base_dir = Path(user_cache_dir(PACKAGE_NAME))
    return _get_env_path("VIBEUSAGE_CACHE_DIR", base_dir)


def state_dir() -> Path:
    """Get user state directory for runtime data."""
    return Path(user_state_path(PACKAGE_NAME))


def credentials_dir() -> Path:
    """Get credentials subdirectory."""
    return config_dir() / "credentials"


def snapshots_dir() -> Path:
    """Get cached snapshots directory."""
    return cache_dir() / "snapshots"


def org_ids_dir() -> Path:
    """Get cached org IDs directory."""
    return cache_dir() / "org-ids"


def gate_dir() -> Path:
    """Get failure gate state directory."""
    return state_dir() / "gates"


def config_file() -> Path:
    """Get main config.toml path."""
    return config_dir() / "config.toml"


def ensure_directories() -> None:
    """Ensure all required directories exist."""
    for directory in (
        config_dir(),
        cache_dir(),
        state_dir(),
        credentials_dir(),
        snapshots_dir(),
        org_ids_dir(),
        gate_dir(),
    ):
        directory.mkdir(parents=True, exist_ok=True)

File: vibeusage-go/internal/config/paths.go
package config

import (
	"os"
	"path/filepath"
	"runtime"
)

const appName = "vibeusage"

func ConfigDir() string {
	if v := os.Getenv("VIBEUSAGE_CONFIG_DIR"); v != "" {
		return v
	}
	return filepath.Join(userConfigDir(), appName)
}

func CacheDir() string {
	if v := os.Getenv("VIBEUSAGE_CACHE_DIR"); v != "" {
		return v
	}
	return filepath.Join(userCacheDir(), appName)
}

func StateDir() string {
	return filepath.Join(userStateDir(), appName)
}

func CredentialsDir() string { return filepath.Join(ConfigDir(), "credentials") }
func SnapshotsDir() string   { return filepath.Join(CacheDir(), "snapshots") }
func OrgIDsDir() string      { return filepath.Join(CacheDir(), "org-ids") }
func GateDir() string        { return filepath.Join(StateDir(), "gates") }
func ConfigFile() string     { return filepath.Join(ConfigDir(), "config.toml") }

func EnsureDirectories() error {
	dirs := []string{
		ConfigDir(), CacheDir(), StateDir(),
		CredentialsDir(), SnapshotsDir(), OrgIDsDir(), GateDir(),
	}
	for _, d := range dirs {
		if err := os.MkdirAll(d, 0o755); err != nil {
			return err
		}
	}
	return nil
}

func userConfigDir() string {
	if d, err := os.UserConfigDir(); err == nil {
		return d
	}
	return filepath.Join(homeDir(), ".config")
}

func userCacheDir() string {
	if d, err := os.UserCacheDir(); err == nil {
		return d
	}
	return filepath.Join(homeDir(), ".cache")
}

func userStateDir() string {
	switch runtime.GOOS {
	case "darwin":
		return filepath.Join(homeDir(), "Library", "Application Support")
	case "windows":
		if d := os.Getenv("LOCALAPPDATA"); d != "" {
			return d
		}
		return filepath.Join(homeDir(), "AppData", "Local")
	default:
		if d := os.Getenv("XDG_STATE_HOME"); d != "" {
			return d
		}
		return filepath.Join(homeDir(), ".local", "state")
	}
}

func homeDir() string {
	if d, err := os.UserHomeDir(); err == nil {
		return d
	}
	return "."
}
